backend/src/nodes/nodes/utility/round.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import math
+from enum import Enum
+from typing import Union
+
+from ...groups import conditional_group
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import EnumInput, NumberInput
+from ...properties.outputs import NumberOutput
+from . import category as UtilityCategory
+
+
+class RoundOperation(Enum):
+    FLOOR = "Down"
+    CEILING = "Up"
+    ROUND = "Either way"
+
+
+class RoundScale(Enum):
+    UNIT = "Whole Number"
+    MULTIPLE = "Multiple of..."
+    POWER = "Power of..."
+
+
+@NodeFactory.register("chainner:utility:math_round")
+class RoundNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Round an input number"
+        self.inputs = [
+            NumberInput(
+                "Input",
+                minimum=None,
+                maximum=None,
+                precision=100,
+                controls_step=1,
+            ),
+            EnumInput(
+                RoundOperation,
+                "Round",
+                option_labels={k: k.value for k in RoundOperation},
+            ),
+            EnumInput(
+                RoundScale,
+                "to the nearest",
+                option_labels={k: k.value for k in RoundScale},
+            ),
+            conditional_group(enum=2, condition=RoundScale.MULTIPLE.value)(
+                NumberInput(
+                    "Multiple",
+                    default=1,
+                    minimum=1e-100,
+                    maximum=None,
+                    precision=100,
+                    controls_step=1,
+                )
+            ),
+            conditional_group(enum=2, condition=RoundScale.POWER.value)(
+                NumberInput(
+                    "Power",
+                    default=2,
+                    minimum=(1 + 1e-100),
```

Comment:
Floating point imprecision.

```py
>>> 1 + 1e-100
1.0
```

Reply:
Oops.  Yeah, I should have realized that.  I guess `np.nextafter(1.0, np.inf)` is the technically right way to do this.  I wonder if a better minimum for any practical use would be `2`.